react/lib/util/api-client.ts

```diff
@@ -1,9 +1,41 @@
 import axios from 'axios';
 import _ from 'lodash';
-import { Socket } from 'socket.io-client'
 import config from '../config.json'
 import { isValidCashAddress, isValidXecAddress } from './address';
-
+import {
+  Transaction,
+  UtxoDetails,
+  PriceData,
+  TransactionDetails,
+  currency
+} from './types';
+import { isFiat } from './currency';
+import { resolveNumber } from './number';
+
+
+export const shouldTriggerOnSuccess = (
+  transaction: Transaction,
+  thisPaymentId?: string,
+  cryptoAmount?: string,
+  thisOpReturn?: string,
+) => {
+  const { amount, paymentId, message } = transaction;
+
+  const formattedTxAmount = resolveNumber(amount);
+  const formattedAmountSet = cryptoAmount ? resolveNumber(cryptoAmount) : false;
+  const formattedTxOpReturn = JSON.stringify(message);
```

Comment:
why are we JSON.stringifying these strings?

Reply:
for some reason sometimes the string is interpreted as an object so I decided to stringfy both to compare. 
we do it with both value because if we compare empty strings without stringfy it will compare this "''" with ''  so we need to do it for both variables